src/txdb.cpp

```diff
@@ -124,7 +124,7 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock, boo
             changed++;
         }
         count++;
-        it = erase ? mapCoins.erase(it) : std::next(it);
+        it = cursor.Next(erase);
```

Comment:
nit: does the update have to be in the middle or can it go to the top, like the rest (the related tests pass that way)?

Reply:
I kept this here intentionally to make the diff easier to review. We don't have to consider whether it breaks anything if we move it up.